assets/src/edit-story/components/inspector/inspectorProvider.js

```diff
@@ -6,26 +6,78 @@ import PropTypes from 'prop-types';
 /**
  * WordPress dependencies
  */
-import { useState } from '@wordpress/element';
+import { useCallback, useState } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
+import { useAPI } from '../../app/api';
 import Context from './context';
 
 const DESIGN = 'design';
 const DOCUMENT = 'document';
 const PREPUBLISH = 'prepublish';
 
 function InspectorProvider( { children } ) {
+	const { actions: { getAllStatuses, getAllUsers } } = useAPI();
 	const [ tab, setTab ] = useState( DESIGN );
+	const [ users, setUsers ] = useState( [] );
+	const [ statuses, setStatuses ] = useState( [] );
+
+	const [ isUsersLoading, setIsUsersLoading ] = useState( false );
+	const [ isStatusesLoading, setIsStatusesLoading ] = useState( false );
+	const [ disabledStatuses, setDisabledStatuses ] = useState( [] );
+
+	const loadStatuses = useCallback( () => {
+		if ( ! isStatusesLoading && statuses.length === 0 ) {
+			setIsStatusesLoading( true );
+			getAllStatuses().then( ( data ) => {
+				data = Object.values( data );
+				// eslint-disable-next-line camelcase
+				data = data.filter( ( { show_in_list } ) => show_in_list );
+				const saveData = data.map( ( {
+					slug,
+					name,
+				} ) => ( {
+					value: slug,
+					name,
+				} ) );
+				setStatuses( saveData );
+				setIsStatusesLoading( false );
+			} );
+		}
+	}, [ isStatusesLoading, statuses.length, getAllStatuses ] );
+
+	const loadUsers = useCallback( () => {
+		if ( ! isUsersLoading && users.length === 0 ) {
+			setIsUsersLoading( true );
+			getAllUsers().then( ( data ) => {
+				const saveData = data.map( ( {
+					id,
+					name,
+				} ) => ( {
+					value: id,
+					name,
+				} ) );
+
+				setUsers( saveData );
+				setIsUsersLoading( false );
+			} );
+		}
+	}, [ isUsersLoading, users.length, getAllUsers ] );
```

Comment:
These two callbacks are also very similar now and could be baked into the above-mentioned reducer-based hooks, that will manage their loading. Will make this much cleaner.